static/app/views/issueDetails/streamline/foldSection.tsx

```diff
@@ -0,0 +1,84 @@
+import {useCallback} from 'react';
+import styled from '@emotion/styled';
+
+import {IconChevron} from 'sentry/icons';
+import {space} from 'sentry/styles/space';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+const LOCAL_STORAGE_PREFIX = 'issue-details-collapse-';
+export const enum SectionKey {
+  HIGHLIGHTS = 'highlights',
+  STACK_TRACE = 'stack-trace',
+  BREADCRUMBS = 'breadcrumbs',
+  TAGS = 'tags',
+  CONTEXTS = 'contexts',
+}
+
+interface FoldSectionProps {
+  children: React.ReactNode;
+  sectionKey: SectionKey;
+  title: React.ReactNode;
+  initialCollapse?: boolean;
+  preventCollapse?: boolean;
+}
+
+export function FoldSection({
+  children,
+  title,
+  sectionKey,
+  initialCollapse = false,
+  preventCollapse = false,
+  ...props
+}: FoldSectionProps) {
+  const [isCollapsed, setIsCollapsed] = useLocalStorageState(
+    `${LOCAL_STORAGE_PREFIX}${sectionKey}`,
+    initialCollapse
+  );
+
+  const toggleCollapse = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+      setIsCollapsed(!isCollapsed);
```

Comment:
Should use the callback when the state change is dependent on the old state: `setIsCollapsed(collapsed => !collapsed)`